src/container/srv_container.c

```diff
@@ -1981,6 +2046,17 @@ cont_open(struct rdb_tx *tx, struct ds_pool_hdl *pool_hdl, struct cont *cont,
 			DP_CONT(cont->c_svc->cs_pool_uuid, cont->c_uuid), out->coo_lsnapshot);
 	}
 
+	/* Get number of open handles (pre-incremented to reflect the effects of this open) */
+	if (cont_proto_ver >= CONT_PROTO_VER_WITH_NHANDLES) {
+		rc = get_nhandles(tx, cont, NHANDLES_PRE_INCREMENT, &nhandles);
```

Comment:
[Not requesting changes] It looks like we don't need the special "PRE_<OP>" concepts or doing updates in functions named "get_<...>". What would the problem be if we simply write:

    lookup(tx, &nhandles)
    nhandles++ /* increment the variable for preforming the update call */
    update(tx, &nhandles)
    commit(tx)
    if tx cannot commit
        if necessary, do not publish nhandles and decrement the variable if needed

I think we may not even need the last `if` statement for cont_open.

Reply:
I was trying to consolidate the logic for looking up and updating this property across multiple calling locations, rather than copy/paste the code in each location. Agree that the word "get_" in the function can be misleading the way it has been implemented here.